tools/mason/MasonExternal.chpl

```diff
@@ -79,28 +118,108 @@ private proc specHelp() {
+  }
+  if getSpackVersion != spackVersion && SPACK_ROOT == spackDefaultPath {
+    throw new owned MasonError("Mason has been updated and requires a newer" +
+          " version of Spack.\nTo use mason external, call: mason external --setup");
   }
   return true;
 }
 
 /* Spack installed to MASON_HOME/spack */
 proc setupSpack() throws {
   writeln("Installing Spack backend ...");
-  const destination = MASON_HOME + "/spack/";
-  const clone = "git clone -q https://github.com/spack/spack " + destination;
-  const checkout = "git checkout -q " + spackVersion;
-  const status = runWithStatus(clone);
-  gitC(destination, checkout);
-  if status != 0 {
+  const destCLI = MASON_HOME + "/spack/";
+  const spackLatestBranch = ' --branch v' + spackVersion + ' ';
+  const destPackages = MASON_HOME + "/spack-registry";
+  const spackMasterBranch = ' --branch master ';
+  const statusCLI = cloneSpackRepository(spackLatestBranch, destCLI);
+  const statusPackages = cloneSpackRepository(spackMasterBranch, destPackages);
+  generateYAML();
+  if statusCLI != 0 && statusPackages != 0 {
     throw new owned MasonError("Spack installation failed");
   }
 }
 
+/* Clones the Spack repository */
+proc cloneSpackRepository(branch : string, dest: string) {
+  const repo = "https://github.com/spack/spack ";
+  const depth = '--depth 1 ';
+  const command = 'git clone -q -c advice.detachedHead=false ' + branch + depth + repo + dest;
+  const statusPackages = runWithStatus(command);
+  if statusPackages != 0 then return -1;
+  else return 0;
+}
+
+/* git checkout command run at SPACK_ROOT */
+proc gitCheckOutSpack(tag: string) {
+  const checkOutCommand = 'git ' + '-C ' + SPACK_ROOT +
+                      ' checkout -q ' + tag;
+  const status = runWithStatus(checkOutCommand);
+  if status != 0 then return -1;
+  else return 0;
+}
+
+/* git fetch command run at SPACK_ROOT */
+proc gitFetch(branch: string) {
+  const command = 'git ' + '-C ' + SPACK_ROOT + ' fetch -q ' +
+                  ' --depth 1 origin ' + branch;
+  const status = runWithStatus(command);
+  if status != 0 then return -1;
+  else return 0;
+}
+
+/* Updates the spack directory used for spack commands */
+private proc updateSpackCommandLine() {
+  const releaseTag = 'v' + spackVersion;
+  var tag = 'refs/tags/' + releaseTag;
+  tag = tag + ':' + tag;
+  const statusFetch = gitFetch(tag);
+  const statusCheckOut = gitCheckOutSpack(releaseTag);
+  if statusFetch != 0 || statusCheckOut != 0 then return -1;
+  else return 0;
+}
+
+/*
+  Generate site-specific repos.yaml :-
+  Replaces package repository path of repos.yaml file at
+  /spack/etc/spack/defaults with that of spack-registry
+*/
+private proc generateYAML() {
+  const yamlFilePath = SPACK_ROOT + '/etc/spack/defaults/repos.yaml';
+  if isFile(yamlFilePath) {
+    remove(yamlFilePath);
+  }
+  const reposOverride = 'repos:\n'+
+                        '  - ~/.mason/spack-registry/var/spack/repos/builtin \n';
```

Comment:
@ankingcodes - in doing some testing for removing the SKIPIF, I ran into an issue here -- this should be `MASON_HOME` instead of `~/.mason`.

Reply:
@ben-albrecht  Okay, I'll make a PR for this.

Reply:
Thanks!